File: 0143-reorder-list/0143-reorder-list.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
void reorderList(struct ListNode* head) {
    struct ListNode* slow = head;
    struct ListNode* fast = head;
    struct ListNode* p = head;
    while(fast!=NULL&&fast->next!=NULL){
        slow = slow->next;
        fast = fast->next->next;
    }
    struct ListNode* secound = slow->next;
    slow->next = NULL;

    struct ListNode* prev = NULL;
    while(secound){
        struct ListNode* next = secound->next;
        secound->next = prev;
        prev = secound;
        secound = next;
    }

    struct ListNode* first = head;
    secound = prev;
    while(secound){
        struct ListNode* t1 = first->next;
        struct ListNode* t2 = secound->next;
        first->next = secound;
        secound->next = t1;
        first = t1;
        secound = t2;
    }
    
    
}

File: 0143-reorder-list/0143-reorder-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def reorderList(self, head: Optional[ListNode]) -> None:
        """
        Do not return anything, modify head in-place instead.
        """
        fast = head
        slow = head

        while fast.next and fast.next.next:
            slow = slow.next
            fast = fast.next.next
        temp = slow.next
        slow.next = None
        
        
        
        prev = None
        current = temp
        while current:
            nxt = current.next
            current.next = prev
            prev = current
            current = nxt
        new = head
        new1 = prev

        while new and new1:
            temp = new.next
            temp1 = new1.next
            new.next = new1
            
            new1.next = temp

            new = temp
            new1 = temp1
            